evergreen/eg_app/views.py

```diff
@@ -1,8 +1,59 @@
+# Create your views here
+#views are just the handler for shit that will be pass into urls.py
+
+
 from django.shortcuts import render
-from django.http import HttpResponse
+from django.http import HttpResponse, HttpResponseNotFound
+from django.conf import settings
+import mimetypes
+from pathlib import Path
+from urllib.parse import quote
+
+
+
+def getFileType(filePath):
+    return mimetypes.guess_type(filePath)[0]  
+
+def indexHandler(request):
+    return render(request, 'index.html')
+
+def fileHandler(request, fileName): # can handle img and text 
+    sanitizedfileName = quote(fileName) # get a clean full file path 
+    path = Path(settings.STATIC_ROOT) / sanitizedfileName
+
+    allowedType = {'.css', '.html', '.js', '.png', '.jpg', '.jpeg', '.gif', '.mp3', '.mp4', '.xml', '.json', '.pdf'} # can add more, 
+
+    if not path.suffix.lower() in allowedType: # to deal with user uploads 
+        return HttpResponseNotFound("404 - File type not allowed")
+    
+    #deal with /../ attacks 
+    rootPath = Path(settings.STATIC_ROOT).resolve() #.resolve get absolute path 
+    if not path.resolve().is_relative_to(rootPath):
+        return HttpResponseNotFound("404 - Invalid file path")
+    
+    if path.exists() and path.is_file(): # make sure its not a directory 
+        contentType = getFileType(path)
+        
+        try:
+            with open(path, 'rb') as file:
+                content = file.read()
 
-# Create your views here.
+            response = HttpResponse(content, content_type=contentType)
+            response['X-Content-Type-Options'] = "nosniff"
+            response['Content-Length'] = str(len(content))
```

Comment:
I haven't looked at the docs but do we really need to set Content-Length ourselves? That really seems like something that should just be handled automatically as part of the library

Reply:
This should be handled by Django, yeah you're right

Reply:
it is handle by django, but django sometime get the wrong len when dealing with the large file, will be commented out since wont make difference rn